lib/spack/spack/cmd/diy.py

```diff
@@ -91,13 +101,21 @@ def diy(self, args):
         tty.msg("Uninstall or try adding a version suffix for this DIY build.")
         sys.exit(1)
 
+    source_path = None
```

Comment:
You don't actually need this line -- ifs in Python [don't have their own scope](https://stackoverflow.com/questions/291978/short-description-of-scoping-rules).